airflow/dag/serialization/schema.json

```diff
@@ -0,0 +1,196 @@
+          "type": "array",
+          "items": { "type": "string" }
+        }
+      },
+      "required": [
+        "__type",
+        "__var"
+      ]
+    },
+    "dag": {
+      "type": "object",
+      "properties": {
+        "params": { "$ref": "#/definitions/dict" },
+        "_dag_id": { "type": "string" },
+        "task_dict": {  "$ref": "#/definitions/task_dict" },
+        "timezone": { "$ref": "#/definitions/timezone" },
+        "schedule_interval": {
+          "anyOf": [
+            { "type": "null" },
+            { "type": "string" },
+            { "$ref": "#/definitions/typed_timedelta" },
+            { "$ref": "#/definitions/typed_relativedelta" }
+          ]
+        },
+        "catchup": { "type": "boolean" },
+        "is_subdag": { "type": "boolean" },
+        "fileloc": { "type" : "string"},
+        "orientation": { "type" : "string"},
+        "_description": { "type" : "string"},
+        "_concurrency": { "type" : "number"},
+        "max_active_runs": { "type" : "number"},
+        "default_args": { "$ref": "#/definitions/dict" },
+        "start_date": { "$ref": "#/definitions/datetime" },
+        "dagrun_timeout": { "$ref": "#/definitions/timedelta" },
+        "doc_md": { "type" : "string"}
+      },
+      "required": [
+        "params",
+        "_dag_id",
+        "fileloc",
+        "task_dict"
+      ],
+      "additionalProperties": false
+    },
+    "task_dict": {
+      "type": "object",
+      "additionalProperties": { "$ref": "#/definitions/operator" }
+    },
+    "operator": {
+      "$comment": "A task/operator in a DAG",
+      "type": "object",
+      "required": [
+        "_task_type",
+        "_task_module",
+        "task_id",
+        "ui_color",
+        "ui_fgcolor",
+        "template_fields"
+      ],
+      "properties": {
+        "_task_type": { "type": "string" },
+        "_task_module": { "type": "string" },
+        "task_id": { "type": "string" },
+        "owner": { "type": "string" },
+        "start_date": { "$ref": "#/definitions/datetime" },
+        "end_date": { "$ref": "#/definitions/datetime" },
+        "trigger_rule": { "type": "string" },
+        "depends_on_past": { "type": "boolean" },
+        "wait_for_downstream": { "type": "boolean" },
+        "retries": { "type": "number" },
+        "queue": { "type": "string" },
+        "pool": { "type": "string" },
+        "execution_timeout": { "$ref": "#/definitions/timedelta" },
+        "retry_delay": { "$ref": "#/definitions/timedelta" },
+        "retry_exponential_backoff": { "type": "boolean" },
+        "params": { "$ref": "#/definitions/dict" },
+        "priority_weight": { "type": "number" },
+        "weight_rule": { "type": "string" },
+        "executor_config": { "$ref": "#/definitions/dict" },
+        "do_xcom_push": { "type": "boolean" },
+        "ui_color": { "$ref": "#/definitions/color" },
+        "ui_fgcolor": { "$ref": "#/definitions/color" },
+        "template_fields": {
```

Comment:
Why is this not a `stringset` ? The values must be unique and the order does not matter.

Reply:
This type has been previously defined.
https://github.com/apache/airflow/pull/5743/files#diff-c5a9d813838f2de6a2ffef12f84df3ddR76-R88
this type has been previously defined.

Reply:
Oh duh, i forgot about that :)

Mostly because the extra "type deocration" isn't needed anymore and we've removed it in cases where it is un-ambigious to the deserializer. (For the huge dags  we've been testing with the parsing time starts to have a non-trivial impact on performance). Since `stringset` isn't used anymore we should probably remove that definition instead.

Reply:
Removed